### PR TITLE
8343477: Remove unnecessary @SuppressWarnings annotations (compiler)

### PR DESCRIPTION
**Pull request by archiecobbs**
Please review this patch which removes unnecessary `@SuppressWarnings` annotations.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (2 reviews required, with at least 2 [Reviewers](https://openjdk.org/bylaws#reviewer))

### Issue
 * [JDK-8343477](https://bugs.openjdk.org/browse/JDK-8343477): Remove unnecessary @SuppressWarnings annotations (compiler) (**Sub-task** - P5)

### Reviewers
 * [Joe Darcy](https://openjdk.org/census#darcy) (@jddarcy - **Reviewer**)
 * [Maurizio Cimadamore](https://openjdk.org/census#mcimadamore) (@mcimadamore - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21851/head:pull/21851` \
`$ git checkout pull/21851`

Update a local copy of the PR: \
`$ git checkout pull/21851` \
`$ git pull https://git.openjdk.org/jdk.git pull/21851/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21851`

View PR using the GUI difftool: \
`$ git pr show -t 21851`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21851.diff">https://git.openjdk.org/jdk/pull/21851.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21851#issuecomment-2453024402)
</details>

### CODE REVIEW

**Review by jddarcy — APPROVED** [review]
Looks fine; thanks for the cleanup @archiecobbs .

Bumped the reviewer count so someone who more directly works on javac will take a look too.

File: src/jdk.compiler/share/classes/com/sun/tools/javac/model/AnnotationProxyMaker.java [hunk 1]
```diff
@@ -220,7 +220,7 @@ public void visitArray(Attribute.Array a) {
             }
         }
 
-        @SuppressWarnings({"unchecked", "rawtypes"})
+        @SuppressWarnings("unchecked")
```
Comment (mcimadamore) [inline src/jdk.compiler/share/classes/com/sun/tools/javac/model/AnnotationProxyMaker.java:223]: Tricky but correct :-) This relies on the special pleading for casts to "raw" `Class`

**Review by mcimadamore — APPROVED** [review]
Looks good - nice cleanup!

**Pull request closed without merging**